### ISSUE TITLE
Build out websocket support

### ISSUE THREAD

**Issue by striglia** [striglia/stockfighter#9]
See https://starfighter.readme.io/docs/quotes-ticker-tape-websocket for details.

People seem to have had luck w/ the ws4py library in python.

Some sharp edges to watch out for:
- https://discuss.starfighters.io/t/websockets-silently-drop-connection-after-5-min-idle/2663
- https://discuss.starfighters.io/t/quotes-going-quiet-after-a-while/2457

I have no idea how to effectively test any of this, since testing API keys aren't yet a real thing.

**Comment by imrehg** [comment]
One issue I've run into working on this is that `six.moves.urllib.parse.urljoin` does not know about the `wss://` protocol for secure websockets, to be able to use the same (or similar) pattern to construct thr URLs as the HTTP API. Filed a [issue](http://bugs.python.org/issue25895) upstream, and in the meantime (just as a note), and the issue can be worked around with roughly lines:

```
from six.moves.urllib import parse
parse.uses_relative.expand(["wss"])
parse.uses_netlock.expand(["wss"])
```

BTW, ws4py seems to be a good catch.

As for setting things up, I think the WebSocket `recv()` should probably be abstracted away, and have some kind of construct that `yield` results. Though I wonder how would that work in Python 2 and 3 uniformly...

**Comment by striglia** [comment]
Nice catch :) A yield-like interface seems like nice Python sytax sugar, although I don't know a lot about how websocket libraries react to incoming messages....is there an event loop involved?

**Comment by imrehg** [comment]
Yeah, the pattern does matter. Looking at the [ws4py client tutorial](https://ws4py.readthedocs.org/en/latest/sources/clienttutorial/), it works asynchronously, though then the question becomes what form should it take.

Just random musing, some ideas: Could be similar as the demo client - have an class that handles the incoming messages and an instance of it returned. Could also take a callback function on instantiation which is executed on every message. Another tricky part, I think, is to abstract away the connection setup and close....
Not sure, I'm probably a decent hacker, but not good enough architect :P

**Comment by striglia** [comment]
I'd vote for "something that works" ahead of worrying too much about design :) I'm hoping to find some free time the next couple weeks to play more of the actual game(!) which will likely inspire some more focused work on the API.

All that said, I prioritize working code over talk, so I'm very likely to accept the first PR w/ websocket support and we can work the rest out later :)
